pkg/controller/config/config_controller.go

```diff
@@ -214,6 +223,9 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
 		for _, entry := range instance.Spec.Sync.SyncOnly {
 			gvk := schema.GroupVersionKind{Group: entry.Group, Version: entry.Version, Kind: entry.Kind}
 			newSyncOnly.Add(gvk)
+
+			r.configMatchSet.Reset()
```

Comment:
We should make changes to the match set atomic.

Things like `Reset()` could cause namespaces to be briefly un-ignored, leading to unintended enforcement.

Generally I see people do this by swapping the internal storage map with a newly provided collection of data, combined with mutex locks.

Reply:
updated with `Replace()` to swap out data